cirq-core/cirq/work/observable_measurement.py

```diff
@@ -354,6 +355,47 @@ def _to_sweep(param_tuples):
     return to_sweep
 
 
+def _parse_checkpoint_options(
+    checkpoint: bool, checkpoint_fn: Optional[str], checkpoint_other_fn: Optional[str]
+) -> Tuple[Optional[str], Optional[str]]:
+    """Parse the checkpoint-oriented options in `measure_grouped_settings`.
+
+    This function contains the validation and defaults logic. Please see
+    `measure_grouped_settings` for documentation on these args.
+
+    Returns:
+        checkpoint_fn, checkpoint_other_fn: Parsed or default filenames for primary and previous
+            checkpoint files.
+    """
+    if not checkpoint:
+        if checkpoint_fn is not None or checkpoint_other_fn is not None:
+            raise ValueError(
+                "Checkpoint filenames were provided but `checkpoint` was set to False."
+            )
+        return None, None
+
+    if checkpoint_fn is None:
+        checkpoint_dir = tempfile.mkdtemp()
+        chk_basename = 'observables'
+        checkpoint_fn = f'{checkpoint_dir}/{chk_basename}.json'
+
+    if checkpoint_other_fn is None:
+        checkpoint_dir = os.path.dirname(checkpoint_fn)
+        chk_basename = os.path.basename(checkpoint_fn)
+        chk_basename, _, ext = chk_basename.rpartition('.')
+        if ext != 'json':
+            raise ValueError(
```

Comment:
The original intent was to limit the amount of "magic" it would do for you, and if you're not following normal filename semantics then there's no telling what's going on. Specifically: if you don't have a file extension `rpartition` will give you `('', '', 'filename'` which would give a weird automatic `checkpoint_other_fn`. 

If you had something like `.tar.gz` the automatic filename would be `basename.tar.prev.gz` which is also weird. 

Do you think I should relax it? I could meet halfway and accept other file extensions but reject the case where there's no file extension, i.e. there's no `'.'` in the name.

Reply:
I added some validation that it follows `filname.ext` pattern. I left in the json extension check. Let me know what you think and I can remove the json extension check.

Reply:
No strong opinions on my end either, but your comment helped clarify the reasoning for me. I think the current behavior should be fine.

Reply:
ok keeping the current behavior. We can loosen the json extension check further if the need arises